### PR TITLE
feat: created settings page issues #18 #19

### PR DESCRIPTION
**Pull request by namansharma3007**
I have successfully completed the following updates:

### Built the settings page.
- Resolved issues #18 and #19.
- Ensured all changes adhere to the project’s file structure, with modifications located in the ```/components/settings``` directory.
- Additionally, I have attached a video as proof of functionality.

https://github.com/user-attachments/assets/20b32f5a-c54b-4ed2-b2a6-bf832faf3039

### CODE REVIEW

**Comment by bryanlundberg** [comment]
> @bryanlundberg Please check this PR

I'm on that 👌

**Comment by namansharma3007** [comment]
@bryanlundberg If you like my work I would like to work on other 2 sections as well.
Please create an issue on that.

**Review by bryanlundberg — CHANGES_REQUESTED** [review]
Hey @namansharma3007 UI looks very good 😅

Regarding the code, there are several minor refactoring opportunities, but nothing critical. My only concern is that if a user manually enters an invalid /settings sub-URL, the page stays blank. It would be better if the logic redirected any non-matching sub-URL back to /settings.

image

**Comment by namansharma3007** [comment]
> Hey @namansharma3007 UI looks very good 😅
> 
> Regarding the code, there are several minor refactoring opportunities, but nothing critical. My only concern is that if a user manually enters an invalid /settings sub-URL, the page stays blank. It would be better if the logic redirected any non-matching sub-URL back to /settings.
> 
> image

Would you also like to implement the functionality that if user enters random text, the random text must disappear from URL, or it must stay there??

**Comment by bryanlundberg** [comment]
> > Hey @namansharma3007 UI looks very good 😅
> > Regarding the code, there are several minor refactoring opportunities, but nothing critical. My only concern is that if a user manually enters an invalid /settings sub-URL, the page stays blank. It would be better if the logic redirected any non-matching sub-URL back to /settings.
> > image
> 
> Would you also like to implement the functionality that if user enters random text, the random text must disappear from URL, or it must stay there?? User can also be redirected to page not found.

Redirect to /settings would be fine.

**Comment by namansharma3007** [comment]
> > > Hey @namansharma3007 UI looks very good 😅
> > > Regarding the code, there are several minor refactoring opportunities, but nothing critical. My only concern is that if a user manually enters an invalid /settings sub-URL, the page stays blank. It would be better if the logic redirected any non-matching sub-URL back to /settings.
> > > image
> > 
> > 
> > Would you also like to implement the functionality that if user enters random text, the random text must disappear from URL, or it must stay there?? User can also be redirected to page not found.
> 
> Redirect to /settings would be fine.

I have resolved the isse: https://github.com/ProyectLink/super-list-frontend/pull/40

**Review by bryanlundberg — COMMENTED** [review]
I can't accept changes on package-lock.json and package.json files when it's not necessary,

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "react-dom": "^18.3.1",
     "react-hook-form": "^7.53.0",
     "react-router-dom": "^6.26.1",
+    "super-list-clone": "file:",
```
[the remark this answers is not in the document]
Comment (namansharma3007) [reply package.json:28]: I cloned this repo twice. It comes when I run the project. But does not affect the functionality of code.

[the remark this answers is not in the document]
Comment (namansharma3007) [reply package.json:28]: I can commit new changes after removing this. Will that be ok?

[the remark this answers is not in the document]
Comment (bryanlundberg) [reply package.json:28]: The issue I want to avoid, is for other people that will need to fix conflict on same code line modifications before PR.

**Comment by bryanlundberg** [comment]
foward to #40

**Pull request closed without merging**